### PR TITLE
fix(web): improve Proxy check to avoid SSR problems

### PR DESCRIPTION
**Pull request by flogy**
Fixed https://github.com/ionic-team/capacitor/issues/2850

Solves server rendering issues by checking if `window` is actually defined (which is not the case when doing server rendering).

### CODE REVIEW

**Comment by IljaDaderko** [comment]
Just verified this with capacitor / nextjs build, all good 👍

**Comment by jabczyk** [comment]
I ran across the same issue when using Capacitor and Gatsby.
After this change Gatsby's SSR works perfectly fine.
Please approve it in the next update, as it delays my deployment.

**Comment by IljaDaderko** [comment]
Any eta on when we can merge this @jcesarmobile ?

File: core/src/web-runtime.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class CapacitorWeb {
     // Gracefully degrade in non-Proxy supporting engines, e.g. IE11. This
     // effectively means that trying to access an unavailable plugin will
     // locally throw, but this is still better than throwing a syntax error.
-    if ('Proxy' in window) {
+    if ('Proxy' in (typeof window === 'undefined' ? global : window)) {
```
Comment (dwieeb) [inline core/src/web-runtime.ts:16]:
```suggestion
    if (typeof Proxy !== 'undefined') {
```

Comment (flogy) [reply core/src/web-runtime.ts:16]: Just tested it in the sample project (https://codesandbox.io/s/github/flogy/gatsby-starter-capacitor). Building works fine and it also still runs in IE 11 browser (which was the initial problem fixed by https://github.com/ionic-team/capacitor/pull/2759).

**Comment by fildidi** [comment]
When should we expect this fix to be released? Because currently im having the same issue with Nuxt js, where Im on 'universal' mode, where since the first page need to be served from the server, the window is not defined.
It is for a site, which is suppose to go live within a week, so I urgently need this fix released, so we can us it
Do you have a time frame in mind, on when can that happen?

**Comment by flogy** [comment]
Found an issue with this on IE 11:

The committed code is correct and functional. However, after it was built and released it again breaked IE 11 compatibility (which was originally fixed by #2759). It seems that  building the code adds `''` around the `Proxy` variable:

This is the line of code as committed in this PR:
https://github.com/ionic-team/capacitor/blob/7afc9eb81ad622282a27b6dd3fca30165bf2af95/core/src/web-runtime.ts#L16

This is the same line of code in my `node_module` dependencies:
```javascript
if (typeof 'Proxy' !== "undefined") {
```

When running the if statement in IE 11 browser it returns the following:

```
typeof 'Proxy' !== "undefined"
true

typeof Proxy !== "undefined"
false
```

It should be `false` but is not after packaging and distributing the `@capacitor/core` package.

Could someone experienced in the used packagers check on this issue?

Thanks in advance.

**Pull request merged**